pkg/transport/transport.go

```diff
@@ -36,12 +37,15 @@ type Client interface {
 	DialNode(ctx context.Context, node *pb.Node, opts ...grpc.DialOption) (*grpc.ClientConn, error)
 	DialAddress(ctx context.Context, address string, opts ...grpc.DialOption) (*grpc.ClientConn, error)
 	Identity() *provider.FullIdentity
+	AddObserver(obs Observer)
+	Observers() []Observer
 }
 
 // Transport interface structure
 type Transport struct {
 	identity  *provider.FullIdentity
 	observers []Observer
+	sync.Mutex
```

Comment:
Don't expose the mutex and order such that it can be understood what it is protecting.